src/SFA.DAS.EAS.Account.Worker/Infrastructure/TriggeredJobs.cs

```diff
@@ -0,0 +1,70 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Azure.WebJobs.Host;
+using SFA.DAS.EAS.Account.Worker.Infrastructure.Interfaces;
+using SFA.DAS.EAS.Account.Worker.Jobs;
+
+namespace SFA.DAS.EAS.Account.Worker.Infrastructure
+{
+	public class TriggeredJobs
+	{
+		public void ProcessQueueMessage([QueueTrigger(Constants.AzureQueueNames.AdHocJobQueue)] AdHocJobParams jobParams, TraceWriter logger)
+		{
+			// find implementation of IJob named jobParams.JobName
+			logger.Info($"Received request to process job type {jobParams.JobName}");
+			var jobtype = GetAllPossibleJobs()
+				.FirstOrDefault(jobType => string.Equals(jobType.Name, jobParams.JobName, StringComparison.InvariantCultureIgnoreCase));
+
+			if (jobtype != null)
+			{
+				RunJob(jobtype, logger);
+			}
+		}
+
+		//                                    {second} {minute} {hour} {day} {month} {day-of-week}
+		public void PaymentCheck([TimerTrigger("0 30 5 1 * *")] TimerInfo timer, TraceWriter logger)
+		{
+			ScheduleJob<PaymentIntegrityCheckerJob>();	
+		}
+
+		private void RunJob(Type jobtype, TraceWriter logger)
+		{
+		    using (var thisContainer = ServiceLocator.CreateNestedContainer())
+		    {
+		        ServiceLocator.Register<TraceWriter>(thisContainer, logger);
```

Comment:
I don't think the nested container instance is being used in this call. Are we ok to remove it?

Reply:
Done